routes/documents.js

```diff
@@ -2,56 +2,61 @@ const express = require('express');
 
 const router = express.Router();
 const Document = require('../models/document');
+const User = require('../models/user');
 const { DocumentBodySchema } = require('./validators/validators');
 const validator = require('express-joi-validator');
 
-router.get('/', (req, res, next) => {
-  Document.find({})
-    .select('-content')
-    .then((document) => {
-      res.send(document);
+const sendError = res =>
+  res.status(401).send({
+    message: 'User is not logged!',
+    status: 401,
+  });
+
+
+const isAuthenticated = (req, res, next) =>
+  User.findById(req.signedCookies['simul-doc'])
+    .then((user) => {
+      if (!user) return sendError(res);
+      return next();
     })
     .catch(next);
-});
 
-router.get('/:docId', (req, res, next) => {
-  Document.findById(req.params.docId)
-    .populate('owner', '-createdAt -updatedAt')
-    .populate('authors', '-createdAt -updatedAt')
+router.get('/', isAuthenticated, (req, res, next) =>
+  Document.find({ authors: req.signedCookies['simul-doc'] })
+    .select('-content')
     .then((document) => {
       res.send(document);
     })
-    .catch(next);
-});
+    .catch(next));
 
-router.post('/', validator(DocumentBodySchema), async (req, res, next) => {
-  const id = await Document.create(req.body)
-    .then(document => document.id)
-    .catch(next);
-  Document.findById(id)
+
+router.get('/:docId', isAuthenticated, (req, res, next) =>
+  Document.findById(req.params.docId)
     .populate('owner', '-createdAt -updatedAt')
     .populate('authors', '-createdAt -updatedAt')
-    .then((document) => {
-      res.send(document);
-    })
-    .catch(next);
-});
+    .then(document => res.send(document))
+    .catch(next));
+
+router.post('/', isAuthenticated, validator(DocumentBodySchema), (req, res, next) =>
+  Document.create(req.body)
+    .then(createdDocument =>
+      Document.findById(createdDocument.id)
+        .populate('owner', '-createdAt -updatedAt')
+        .populate('authors', '-createdAt -updatedAt')
+        .then(document => res.send(document))
+        .catch(next))
+    .catch(next));
 
-router.delete('/:docId', (req, res, next) => {
+router.delete('/:docId', isAuthenticated, (req, res, next) =>
   Document.findByIdAndRemove(req.params.docId)
-    .then(() => {
-      res.sendStatus(204);
-    })
-    .catch(next);
-});
+    .then(() => res.sendStatus(204))
+    .catch(next));
 
-router.put('/:docId', validator(DocumentBodySchema), (req, res, next) => {
-  Document.findByIdAndUpdate(req.params.docId, req.body, { new: true })
+router.put('/:docId', isAuthenticated, validator(DocumentBodySchema), (req, res, next) =>
+  Document
```

Comment:
I think that you're not checking here if you're editing a document belonging to a current user. Anyone with document ID could edit it without being a co-author.